src/nix/develop.cc

```diff
@@ -408,9 +409,34 @@ struct CmdDevelop : Common, MixEnvironment
         }
 
         // If running a phase, don't want an interactive shell running after
-        // Ctrl-C, so don't pass --rcfile
-        auto args = phase ? Strings{std::string(baseNameOf(shell)), rcFilePath}
-            : Strings{std::string(baseNameOf(shell)), "--rcfile", rcFilePath};
+        // Ctrl-C, so don't pass --rcfile. Also need to chdir since phases
+        // assume in flake directory
+        Strings args;
+        if (phase) {
+            args = Strings{std::string(baseNameOf(shell)), rcFilePath};
+            // chdir if installable is a flake of type git+file or path
+            auto installableFlake = std::dynamic_pointer_cast<InstallableFlake>(installable);
+            if (installableFlake) {
+                auto & attrs = installableFlake->flakeRef.input.attrs;
+                auto type = getStrAttr(attrs, "type");
+                std::string flakeDir;
+                if (type == "git") {
+                    auto url = parseURL(getStrAttr(attrs, "url"));
+                    if (url.scheme == "file") {
+                        flakeDir = url.path;
+                    }
+                } else if (type == "path") {
+                    flakeDir = getStrAttr(attrs, "path");
+                }
```

Comment:
I think this can be replaced by `installableFlake.getLockedFlake()->flake.resolvedRef.input.getSourcePath()` (or something of that taste)

Reply:
Haven't tested it but changed to use `getSourcePath`